src/EPR.Payment.Service.Data.UnitTests/Repositories/PaymentsRepositoryTests.cs

```diff
@@ -0,0 +1,199 @@
+{
+    [TestClass]
+    public class PaymentsRepositoryTests
+    {
+        private Mock<DbSet<Common.Data.DataModels.Payment>> _paymentMock;
+        private Mock<DbSet<Common.Data.DataModels.Lookups.PaymentStatus>> _paymentStatusMock;
+        public PaymentsRepositoryTests()
+        {
+            _paymentMock = MockIPaymentRepository.GetPaymentMock();
+            _paymentStatusMock = MockIPaymentRepository.GetPaymentStatusMock(true);
+        }
+
+        [TestMethod]
+        [AutoMoqData]
+        public async Task InsertPaymentStatusAsync_ValidInput_ShouldReturnGuid(
+            [Frozen] Mock<AppDbContext> _dataContextMock, 
+            [Greedy] PaymentsRepository _mockPaymentsRepository)
+        {
+            //Arrange
+            _dataContextMock.Setup(i => i.Payment).ReturnsDbSet(_paymentMock.Object);
+            _mockPaymentsRepository = new PaymentsRepository(_dataContextMock.Object);
+
+            var newId = Guid.NewGuid();
+            var userId = Guid.NewGuid();
+            var organisationId = Guid.NewGuid();
+            var request = new Common.Data.DataModels.Payment
+            {
+                Id = newId,
+                UserId = userId,
+                OrganisationId = organisationId,
+            };
+
+            //Act
+            Guid result = await _mockPaymentsRepository.InsertPaymentStatusAsync(request);
+
+
+            //Assert
+             result.Should().NotBe(Guid.Empty);
+            _dataContextMock.Verify(c => c.Payment.Add(It.Is<Common.Data.DataModels.Payment>(s =>s.UserId == userId && s.OrganisationId == organisationId)), Times.Once());
+            _dataContextMock.Verify(c => c.SaveChangesAsync(default), Times.Once);
+        }
+
+        [TestMethod]
+        [AutoMoqData]
+        public async Task InsertPaymentStatusAsync_NullEntity_ShouldThrowArgumentException(
+            [Frozen] Mock<AppDbContext> _dataContextMock,
+            [Greedy] PaymentsRepository _mockPaymentsRepository)
+        {
+            //Arrange
+            _dataContextMock.Setup(i => i.Payment).ReturnsDbSet(_paymentMock.Object);
+            _mockPaymentsRepository = new PaymentsRepository(_dataContextMock.Object);
+
+            Common.Data.DataModels.Payment? request = null;
+
+
+            //Act & Assert
+            await _mockPaymentsRepository.Invoking(async x => await x.InsertPaymentStatusAsync(request))
+                .Should().ThrowAsync<ArgumentException>();
+        }
+
+        [TestMethod]
+        [AutoMoqData]
+        public async Task UpdatePaymentStatusAsync_ValidInput_ShouldComplete(
+            [Frozen] Mock<AppDbContext> _dataContextMock,
+            [Greedy] PaymentsRepository _mockPaymentsRepository)
+        {
+            //Arrange
+            _dataContextMock.Setup(i => i.Payment).ReturnsDbSet(_paymentMock.Object);
+            _mockPaymentsRepository = new PaymentsRepository(_dataContextMock.Object);
+
+            var newId = Guid.NewGuid();
```

Comment:
see earlier comment for Frozen attribute and making code readable and precise

Reply:
resolved